src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -602,23 +640,31 @@ private static void assertGVCFHasOnlyOneSample(final String variantPath, final V
         return new TreeMap<>(loadSampleNameMapFile(sampleToFileMapPath));
     }
 
+    /**
+     * write out interval list to file
+     */
+    private void writeIntervalListToFile() {
+        final IntervalList outputList = new IntervalList(getBestAvailableSequenceDictionary());
+        intervals.forEach(i -> outputList.add(new Interval(i.getContig(), i.getStart(), i.getEnd())));
+        final Path intervalListOutputPath = IOUtils.getPath(intervalListOutputPathString);
+        outputList.write(intervalListOutputPath.toFile());
+    }
+
     /**
      * Before traversal, fix configuration parameters and initialize
      * GenomicsDB. Hard-coded to handle only VCF files and headers
      */
     @Override
     public void onTraversalStart() {
-        if (getIntervalsFromExistingWorkspace) {
-            final IntervalList outputList = new IntervalList(getBestAvailableSequenceDictionary());
-            intervals.forEach(i -> outputList.add(new Interval(i.getContig(), i.getStart(), i.getEnd())));
-            final Path intervalListOutputPath = IOUtils.getPath(intervalListOutputPathString);
-            outputList.write(intervalListOutputPath.toFile());
-            return;
-        }
         String workspaceDir = BucketUtils.makeFilePathAbsolute(overwriteCreateOrCheckWorkspace());
         vidMapJSONFile = IOUtils.appendPathToDir(workspaceDir, GenomicsDBConstants.DEFAULT_VIDMAP_FILE_NAME);
         callsetMapJSONFile = IOUtils.appendPathToDir(workspaceDir, GenomicsDBConstants.DEFAULT_CALLSETMAP_FILE_NAME);
         vcfHeaderFile = IOUtils.appendPathToDir(workspaceDir, GenomicsDBConstants.DEFAULT_VCFHEADER_FILE_NAME);
+        if (getIntervalsFromExistingWorkspace) {
+            // intervals may be null if merge-contigs-into-num-partitions was used to create workspace
+            // if so, we need to wait for vid to be generated before writing out interval list
+            return;
```

Comment:
Don't we still want to output the log messages even if we're getting intervals from an existing workspace?  Why does this need to short circuit here?  It seems like we could drop this whole condition because the initializeInputPreloadExecutorService method now guards against null intervals as well.

Reply:
yup - don't need to short circuit anymore as you say. added a log.info and specific to getIntervalsFromExistingWorkspace (we don't touch the jsons or workspace otherwise in that case)